### PR TITLE
Featur/salem/interestes screen

### PR DESCRIPTION
**Pull request by OmaRoiid**

### CODE REVIEW

File: anabeesh/build.gradle [hunk 1]
```diff
@@ -55,6 +55,7 @@ dependencies {
 
     // Dagger
     implementation deps.dagger.lib
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
```
Comment (Muhammadyoussef) [inline anabeesh/build.gradle]: Remove el line da, Android studio bey3mello add automatic dayman.
E7na keda keda 3amlenlo add mn el file beta3 dependencies.gradle

File: anabeesh/build.gradle [hunk 2]
```diff
@@ -71,6 +72,8 @@ dependencies {
     // Testing
     testImplementation deps.test.junit
     androidTestImplementation deps.test.runner
+    //facebook
+    implementation 'com.facebook.android:facebook-android-sdk:[4,5)'
```
Comment (Muhammadyoussef) [inline anabeesh/build.gradle]: Eh lazmet fb fel PR di? E3mellaha delete

File: anabeesh/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -39,6 +39,42 @@
                 android:name="android.support.PARENT_ACTIVITY"
                 android:value=".ui.landingpage.LandingPageActivity" />
         </activity>
+        <!-- Facebook -->
+        <meta-data
+            android:name="com.facebook.sdk.ApplicationId"
+            android:value="@string/facebook_app_id" />
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/AndroidManifest.xml]: Delete fb, malhash 3elaqa bel PR di

Comment (Muhammadyoussef) [reply anabeesh/src/main/AndroidManifest.xml]: Delete ay 7aga malhash 3elaqa b shoghlak da

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/DataItem.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.cs18.anabeesh.Adapters;
+
+/**
+ TODO: Add class header
+ */
+
+public class DataItem {
+
+    private final Integer img;
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/Adapters/DataItem.java]: Leeh Integer?

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+import java.util.ArrayList;
+
+/**
+ TODO: Add class header
+ */
+
+public class IntrastesAdapter extends RecyclerView.Adapter<IntrastesAdapter.MyViewHolder> {
+    private final ArrayList<DataItem> items;
+    interestesPresenter presenter;
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: Da hayedeelak NullPointerException, lazem te3mello inject fel constructor w e3mel `@Inject` 3ala el constructor

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    Context mContect;
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: E3mel `private` `final`, dayman e3mel final 3ala ad mate2dar (lw msh me7tag teghayaro ba3den) a7san lel memory

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    private int itemClaickableCounter = 0;
+    private boolean positionSelected = false;
+
+    public IntrastesAdapter(@ForActivity Context mContect, ArrayList<DataItem> items) {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: mayenfa3sh te3mel inject lel array list, mate3melhash pass fel parameters beta3et el constructor
E3mellaha initialization gowa el constructor b `new` 3adi

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java (hunk 6 above)
Comment (Muhammadyoussef) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: W lw msh me7tag el reference lel context sheelo lw msh mestakhdemo

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+        View view;
+        ImageView IntrastesImages;
+
+        public MyViewHolder(View itemView) {
+
+            super(itemView);
+            view = itemView;
+            IntrastesImages = itemView.findViewById(R.id.intraestes_Images_id);
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: Leeh msy mestakhdem butter knife?
E3mel bind lel views w estakhdem `ButterKnife.bind(itemView, this)` gowa el constructor beta3 MyViewHolder

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/images.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.cs18.anabeesh.Adapters;
+
+import com.cs18.anabeesh.R;
+
+/**
+ TODO: Add class header
+ */
+
+public class images {
+    public static Integer image[] = {R.mipmap.ic_launcher, R.mipmap.ic_launcher, R.mipmap.ic_launcher, R.mipmap.ic_launcher};
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/Adapters/images.java]: Eh lazmet el class di?

File: anabeesh/src/main/java/com/cs18/anabeesh/AnabeeshApplication.java [hunk 10]
```diff
@@ -32,6 +40,7 @@ public void onCreate() {
         appComponent.inject(this);
         setStrictModeForDebugEnabled(true);
         setTimberDebugTreeEnabled(true);
+        printMyHashKey(this);
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/AnabeeshApplication.java]: Delet unrelated work to your PR

File: anabeesh/src/main/java/com/cs18/anabeesh/model/User.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
+package com.cs18.anabeesh.model;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ TODO: Add class header
+ */
+
+public class User {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/model/User.java]: Leeh el class di?
Leeh 3amel class lel parsing beta3 el user bas?

File: anabeesh/src/main/java/com/cs18/anabeesh/store/FacebookStore.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+import javax.inject.Inject;
+
+import io.reactivex.Single;
+import io.reactivex.SingleEmitter;
+
+/**
+ TODO: Add class header
+ */
+
+class FacebookStore {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/store/FacebookStore.java]: Delete

File: anabeesh/src/main/java/com/cs18/anabeesh/store/FacebookStore.java [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+    private final Context context;
+
+    private LoginButton loginButton;//TODO reference  for  this 
+    private CallbackManager callbackManager;
+
+    @Inject
+    FacebookStore(@ForActivity Context context) {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/store/FacebookStore.java]: Khali kol 7aga 3al branch beta3ha fel PR beta3etha

File: anabeesh/src/main/java/com/cs18/anabeesh/store/InterestesPageRebo.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+ TODO: Add class header
+ */
+
+public class InterestesPageRebo {
+    private final AnabeeshStore anabeeshStore;
+
+    @Inject
+    public InterestesPageRebo(AnabeeshStore anabeeshStore) {this.anabeeshStore = anabeeshStore;}
+
+    public void MakeApicall() {}
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/store/InterestesPageRebo.java]: Ghayar el esm, ana msh fahem da bey3mel ehh keda

File: anabeesh/src/main/java/com/cs18/anabeesh/store/LandingPageRepo.java [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+import javax.inject.Inject;
+
+import io.reactivex.Single;
+
+/**
+ TODO: Add class header
+ */
+
+public class LandingPageRepo {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/store/LandingPageRepo.java]: DELETE UNRELATED WORK

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/InterestesActivity.java [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+import com.cs18.anabeesh.AnabeeshApplication;
+import com.cs18.anabeesh.R;
+import com.cs18.anabeesh.di.activity.ActivityModule;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class InterestesActivity extends Activity {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/InterestesActivity.java]: Add `@ActivityScope`

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/InterestesActivity.java [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+    @BindView(R.id.intrastaing_btn_id)
+    Button Nextbutton;
+
+    @BindView(R.id.intrasting_imgs_RV_id)
+    RecyclerView imagesRecyclerView;
+    @Inject
+    interestesPresenter presenter;
+
+    IntrastesAdapter intrastesAdapter;
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/InterestesActivity.java]: Mestakhdem da fain? W leeh msh ma3mollo inject?

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/InterestesActivity.java [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_interestes);
+        AnabeeshApplication.getComponent(this)
+                .plus(new ActivityModule(this))
+                .inject(this);
+        ButterKnife.bind(this);
+        getActionBar().setDisplayHomeAsUpEnabled(true);
+        Nextbutton.setClickable(false);
+        presenter.BulidRecyclerView(this, imagesRecyclerView);
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/InterestesActivity.java]: Leeh? El mafrod mate3melsh pass l ay `view` lel `presenter`

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/interestesPresenter.java [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.ArrayList;
+
+import javax.inject.Inject;
+
+/**
+ TODO: Add class header
+ */
+
+public class interestesPresenter {
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/interestesPresenter.java]: Rename to `InterestesPresenter`

Comment (Muhammadyoussef) [reply anabeesh/src/main/java/com/cs18/anabeesh/ui/interestes/interestesPresenter.java]: W e3mel add l `@ActivityScope`

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/landingpage/LandingPageActivity.java [hunk 20]
```diff
@@ -20,11 +35,21 @@
...
 
 @ActivityScope
-public class LandingPageActivity extends Activity {
+public class LandingPageActivity extends Activity implements LandingPageScreen {
+
+    @Inject
+    LandingPagePresenter presenter;
+    @BindView(R.id.btn_facebook)
+    LoginButton loginButton;
+    CallbackManager callbackManager;
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/landingpage/LandingPageActivity.java]: 1- sheel el 7agat el malhash 3elaqa bel PR
2- mayenfa3sh te3mel el call back hena, e3melo fel presenter

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/landingpage/LandingPageActivity.java [hunk 21]
```diff
@@ -20,11 +35,21 @@
...
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        //Facebook
+        FacebookSdk.sdkInitialize(getApplicationContext());
+        AppEventsLogger.activateApp(this);
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/landingpage/LandingPageActivity.java]: Delete

File: anabeesh/src/main/java/com/cs18/anabeesh/ui/landingpage/LandingPageActivity.java [hunk 22]
```diff
@@ -34,6 +59,8 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
 
     @OnClick(R.id.btn_facebook)
     void continueWithFacebook() {
+         presenter.continueWithFacebook();
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/java/com/cs18/anabeesh/ui/landingpage/LandingPageActivity.java]: Delete

File: anabeesh/src/main/res/drawable/image_highlight.xml [hunk 23]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <corners android:radius="0dp" />
+    <solid android:color="@android:color/transparent" />
+    <stroke
+        android:width="3dp"
+        android:color="#FFC830" />
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/res/drawable/image_highlight.xml]: Avoid hard coded values, kol 7aga fel res file beta3ha

File: anabeesh/src/main/res/drawable/selector_items_selcted.xml [hunk 24]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/ItemSelected" android:state_pressed="true"
+
+        ></item>
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/res/drawable/selector_items_selcted.xml]: Matensash te3mel `ctrl+alt+O` w `ctrl+alt+L` dayman!

Matsebsh el code bel manzar da

File: anabeesh/src/main/res/layout/activity_interestes.xml [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+    tools:context="com.cs18.anabeesh.ui.interestes.InterestesActivity">
+
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="16dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="16dp"
```
Comment (Muhammadyoussef) [inline anabeesh/src/main/res/layout/activity_interestes.xml]: AVOID HARD CODED VALUES

**Review by Muhammadyoussef — CHANGES_REQUESTED** [review]

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/images.java (hunk 9 above)
Comment (OmaRoiid) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/images.java]: el class de  feh el images ely kont  bgrab 3leha  bs  keda  bs  hya  hattshal  aslan odaam  3shan el apis

Comment (Muhammadyoussef) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/images.java]: E3mel //TODO 3ashan matensash

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/DataItem.java (hunk 4 above)
Comment (OmaRoiid) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/DataItem.java]: 3shan fl Adapter el method   image.setImageResource( "int")  bta5oad  int

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java (hunk 5 above)
Comment (OmaRoiid) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: Kont 3wez A3rf  leeh mmkn  t3ml NPE    m3 eny lma  3mlt run  ma7salsh Keda

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/DataItem.java (hunk 4 above)
Comment (Muhammadyoussef) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/DataItem.java]: Change to 
```
@DrwableRes
private final int img;
```

File: anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java (hunk 5 above)
Comment (Muhammadyoussef) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: 3ashan enta mastakhdemtsh el presenter reference

Comment (Muhammadyoussef) [reply anabeesh/src/main/java/com/cs18/anabeesh/Adapters/IntrastesAdapter.java]: Garrab testakhdemo hayermi null pointer exception

**Pull request closed without merging**